### ISSUE TITLE
ORFik

### ISSUE THREAD

**Issue by JokingHero** [Bioconductor/Contributions#718]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/JokingHero/ORFik

Confirm the following by editing each check box to '[x]'

- [ x ] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [ x ] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [ x ] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [ x ] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [ x ] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [ x ] The 'devel' branch for new packages and features.
- [ x ] The stable 'release' branch, made available every six
      months, for bug fixes.
- [ x ] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by JokingHero** [comment]
Hi, I know we are late to the party, but we still hope you will still find the time to review us before the new Bioconductor release.

**Comment by Roleren** [comment]
Warning from function coverageByWindow:

It is similar to GenomicFeatures::coverageByTranscript.

It uses  IRanges:::regroupBySupergroup, which is not exported from IRanges.

So I see three possible ways of fixing this:
1. We make our own version of regroupBySupergroup.
2. regroupBySupergroup gets exported
3.  GenomicFeatures::coverageByTranscript. becomes more versatile, so we can use that directly.

We will try the first option first.

Thank you.

**Comment by Roleren** [comment]
Ok, so we copied IRanges:::regroupBySupergroup to our package.
Then changed it to be a bit faster, if regroupBySupergroup is exported in the future, we just delete our version.

**Comment by nturaga** [comment]
# ORFik review

## R CMD build

ok

## R CMD INSTALL

ok

## DESCRIPTION

ok

## NAMESPACE

ok

## vignette

* Vignette throws a lot of warnings, is this intended? 

## man

ok

## R

* Please check the NOTE's in your build report,

```
* Checking library calls...
    * NOTE: biocLite, install.packages, or update.packages found in R
      files
    utils.R: 126
    utils.R: 128
    utils.R: 130
```

* use `seq_along` or `seq_len`, instead of n:length()

```
for (i in 2:length(L)) {
	acums[i] <-acums[i-1] + acums[i]
}
```

## src

ok

## tests

ok

Good package overall.

**Comment by Roleren** [comment]
Ok, for the bioclite call inside the package, it is an unexported convenience function that basically installs packages from bioC, so you dont need to find the bioC URL all the time.
I will delete this functions.
Thank you.

**Comment by Roleren** [comment]
Will fix some warnings in vignette, then push again.

**Comment by lawremi** [comment]
It's not great that you copied `IRanges:::regroupBySupergroup()`. There is already a TODO to export that function, so maybe @hpages would be open to it. Also, if you've found a way to improve it, it would be great if you could contribute it to the infrastructure.

**Comment by Roleren** [comment]
Yes, our hope was that i would be exported for 3.7, we are open for sugestions. The version we use is not suitable for IRanges, since it restricts input data classes. I think IRanges:::regroupBySupergroup() is good enough for export with some added docs, if you would like us to contribute with anything( like making a draft for docs) on that matter, let us know.

**Comment by JokingHero** [comment]
Hi @hpages 

Here is the link to the gist with potential improvements to the GenomicFeatures::coverageByTranscript,
[https://gist.github.com/JokingHero/3eb1d69c7fe1adbebc2c1cabc48d8ecb](url)

If you will accept those changes we will be very happy and we will not have to use IRanges:::regroupBySupergroup as we basically wrapped around 
GenomicFeatures::coverageByTranscript and created ORFik:::coverageByWindow

If not than please export IRanges:::regroupBySupergroup so that we don't have to copy this function.

**Comment by JokingHero** [comment]
Is there some sort of solution to the warnings that tokay2 shows for us? Example below:

Rd warning: C:/Users/pkgbuild/AppData/Local/Temp/RtmpkTe16y/R.INSTALL13b833e32206/ORFik/man/asTX.Rd:10: file link 'GRangesList' in package 'GenomicRanges' does not exist and so has been treated as a topic

**Comment by Roleren** [comment]
Ok, so adding the Makewars.win did not fix anything for compilation warning on windows, changing documentation to follow Markdown did not change linking warning on windows, not sure how to fix that.
If anyone have an idea, it is much appreciated.
Thank you.

**Comment by Roleren** [comment]
Any more changes you would like ?

Thank you.

**Comment by Roleren** [comment]
Built clean version 0.99.10.  All concerns in review from @nturaga  fixed.

**Comment by Roleren** [comment]
What happend here. There are no .o files, and rproject have always been there.. 
Any idea ?

**Comment by nturaga** [comment]
Hi, please issue another build by bumping the version.

**Comment by lshep** [comment]
I'm looking into the .o files - This is on our end - We implemented a new check for files and it is not working quite as expected  - The ORFik.Rproj  should be removed with `git rm`  as this should not be tracked - The rest can be ignored for now -

**Comment by Roleren** [comment]
But now the project file is gone, so I can't start RPoject from folder anymore, how should we do this now ?
git rm ORFik.Rproj
result -> deleted:    ORFik.Rproj

The the folder is not a R-package anymore ?

**Comment by nturaga** [comment]
The `.Rproj` file doesn't specify if an R folder is a package or not. These files are ‘extras’ that are put there by RStudio to help it manage your package building. You can just ignore these files, they are not part of ‘required’ R package anatomy.

http://bioconductor.org/developers/how-to/buildingPackagesForBioc/

**Comment by Roleren** [comment]
OK, if you dont have any more concerns, we have fixed what we wanted. Looks good ?

**Comment by nturaga** [comment]
May I suggest that they open an issue under GenomicFeatures for this so we keep track of things? Please follow up with @hpages regarding the issue https://github.com/Bioconductor/GenomicFeatures/issues.

I will accept the package as it looks good.
